### PR TITLE
Fill replay memory

### PR DESCRIPTION
**Pull request by HugoSenetaire**
Factorization of evaluation and fill memory

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -19,16 +19,67 @@ def env_step(env, agent, observation, action_op, goal, channel_names, image_size
...
+    print("eval_rewards",eval_rewards)
+    print("episode length", episode_lengths)
+  
+
+
 def train(dfp_agent, env, eval_env, optimizer, scheduler, args, list_opposition):
   agent = list_opposition[0]
   observation = env.reset()
-  done=False
+  done = False
```
Comment (MatthieuMoreau0) [inline train.py]: take it out

File: train.py [hunk 2]
```diff
@@ -19,16 +19,66 @@ def env_step(env, agent, observation, action_op, goal, channel_names, image_size
...
+            max_score = score
+        GAME += 1
+        score_buffer.append(score)
+        print ("Episode Finished")
+        observation = env.reset()
+        if args['RANDOM_TRAIN_GOAL']:
+          goal = utils.create_goal(list(np.random.rand(len( args['MEASUREMENT_NAMES']))), args["timesteps_goal"])
+    
+    # save the sample <s, a, r, s'> to the replay memory and decrease epsilon
+    dfp_agent.replay_memory(sensory, action_dfp, r_t, None, measurements, is_terminated, explore) # T est demandé mais pourquoi utiliser t ici ? Pas d'utilisation dans replay memory ? Besoin pour multithreading ?
```
Comment (MatthieuMoreau0) [inline train.py]: Obsolete comment

File: default_parameters.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def update_default_args(args):
 
         # Save update
 
-    args["save_every"] = 5000
+    args["save_every"] = 20
```
Comment (MatthieuMoreau0) [inline default_parameters.py]: 20*64 << 5000 ?

Do we want to save that often ?

File: default_parameters.py [hunk 4]
```diff
@@ -82,7 +82,7 @@ def update_default_args(args):
         "frame_per_action" : 1, # TODO : USELESS ?
         "timestep_per_train" : 64,
         "max_memory" : 20000,
-        "agent_evaluate_freq": 5000,
+        "agent_evaluate_freq": 20,
```
Comment (MatthieuMoreau0) [inline default_parameters.py]: same here

**Review by MatthieuMoreau0 — APPROVED** [review]

File: default_parameters.py (hunk 3 above)
Comment (HugoSenetaire) [reply default_parameters.py]: outdated

File: default_parameters.py (hunk 4 above)
Comment (HugoSenetaire) [reply default_parameters.py]: outdated

File: train.py (hunk 1 above)
Comment (HugoSenetaire) [reply train.py]: Ok outdated

File: train.py (hunk 2 above)
Comment (HugoSenetaire) [reply train.py]: Ok

**Review by MatthieuMoreau0 — APPROVED** [review]

**Pull request merged**